docs/wip-plugins.md

```diff
@@ -0,0 +1,140 @@
+# Plugins
+
+If Chatterino is compiled with the `CHATTERINO_PLUGINS` CMake option, it can
+load and execute Lua files. Note that while there are attempts at making this
+decently safe, we cannot guarantee safety.
+
+## Plugin structure
+
+Chatterino searches for plugins in the `Plugins` directory in the app data, right next to `Settings` and `Logs`.
+
+Each plugin should have its own directory.
+
+```
+Chatterino dir/
+└── plugin_name/
+    ├── init.lua
+    └── info.json
+```
+
+`init.lua` will be the file loaded when the plugin is enabled. You may load other files using `loadfile` Lua global function.
+
+`info.json` contains metadata about the plugin, like its name, description,
+authors, homepage link, tags, version, license name. The version field **must**
+be [semver 2.0](https://semver.org/) compliant. The general idea of `info.json`
+will not change however the exact contents probably will, for example with
+permission system ideas.
+Example file:
+
+```json
```

Comment:
Maybe a JSON schema for this could be included, so you can add `"$schema": "https://raw.githubusercontent.com/Chatterino/chatterino2/master/docs/plugin.schema.json"` and get editor feedback.